core/src/main/java/nu/yona/server/goals/service/GoalDTO.java

```diff
@@ -28,20 +30,21 @@
 {
 	private final UUID id;
 	private UUID activityCategoryID;
-	private final ZonedDateTime creationTime;
+	private final Optional<ZonedDateTime> creationTime;
 	private final boolean mandatory;
 
-	protected GoalDTO(UUID id, UUID activityCategoryID, ZonedDateTime creationTime, boolean mandatory)
+	protected GoalDTO(UUID id, Optional<ZonedDateTime> creationTime, UUID activityCategoryID, boolean mandatory)
 	{
+		Objects.requireNonNull(creationTime);
```

Comment:
Can we also verify here that setting the creation time is only allowed in test environments, not on the production environment? I suppose an end user/the app is not allowed to do this?